### DISCUSSION TITLE
Games cataloge project

### DISCUSSION THREAD

**Discussion by talraash** [SilentVoid13/Templater#1069] [Templates Showcase]
I decided to create my own local gaming hub using Obsidian. It is based on Templater and the Obsidian Media DB Plugin. I'm add a few lines of code to media db plugin to parse more parameters from the Steam response. The source code is in my repository, in a fork of the project. The goal, as shown in the diagram below, is to create a highly automated workspace for notes. The idea is to save time that would be spent manually searching for information and formatting, and instead use it for creative pursuits such as notes, reviews, and more.

![image](https://user-images.githubusercontent.com/22260439/227790832-20e29fbe-b7c6-493e-b42b-32023ad8adce.png)

Only the basic scripts and template are ready for now. For example in this script take game id (we use it as name for cssclass) and create or add to main.css file snippet for each notes. This gives us a lot of opportunities. For example, unique animations on each note. Additionally, we don't need to enable each snippet in the Obsidian settings every time. We only need to do it once.

```ts
const fs = require('fs');

function app_add(id_, links_arr) {
  const filePath = '<path_to>/.obsidian/snippets/main.css'; // yep, hardcoded path(i know it's bad but i just begin with js few days ago) to one ccs where we store all snippets 
  const id = 'cs_' + String(id_);
  const img_links = String(links_arr).split(',') // we receive list links with screenshots for background in our note
  const cssCode = `\n\n.${id} {
  -webkit-animation-name: ${id};
  -webkit-animation-duration: 120s;
  -webkit-animation-direction: alternate;
  -webkit-animation-timing-function: linear;
  -webkit-animation-iteration-count: infinite;
}

@-webkit-keyframes ${id} {
  0% {
    background-image: linear-gradient(rgba(0,0,0,0.85), rgba(0,0,0,0.85)), url("${img_links[0]}");
    background-position: center;
    background-size: cover;
  }
  20% {
    background-image: linear-gradient(rgba(0,0,0,0.85), rgba(0,0,0,0.85)), url("${img_links[1]}");
    background-position: center;
    background-size: cover;
  }
  40% {
    background-image: linear-gradient(rgba(0,0,0,0.85), rgba(0,0,0,0.85)), url("${img_links[2]}");
    background-position: center;
    background-size: cover;
  }
  60% {
    background-image: linear-gradient(rgba(0,0,0,0.85), rgba(0,0,0,0.85)), url("${img_links[3]}");
    background-position: center;
    background-size: cover;
  }
  80% {
    background-image: linear-gradient(rgba(0,0,0,0.85), rgba(0,0,0,0.85)), url("${img_links[4]}");
    background-position: center;
    background-size: cover;
  }
  100% {
    background-image: linear-gradient(rgba(0,0,0,0.85), rgba(0,0,0,0.85)), url("${img_links[5]}");
    background-position: center;
    background-size: cover;
  }
}`;

  fs.appendFile(filePath, cssCode, function(err) {
    if (err) {
      console.log('Error appending to file:', err);
    } else {
      console.log('Text added to file successfully!');
    }
  });
}

module.exports = app_add;
```

Main template

```md
---
cssclass: cs_<%tp.frontmatter.id%>
tags: <%tp.frontmatter.genres%>
---
>[!NOTE] Short description
><%tp.frontmatter.short_description%>


> [!FAQ] Quick notes
> - 
> 


| Release date | Played | Metacritic | My Rathing | tags |
| ------------ | ------ | ---------- | ---------- | ---- |
|<%tp.frontmatter.releaseDate%>| True/False|[<%tp.frontmatter.onlineRating%>](<%tp.frontmatter.metacritic_url%>)|10/10|<%tp.frontmatter.genres%>,Game|

<center>
<details>
<summary>Trailers</summary>
<%*
const mov = String(tp.frontmatter.movies).split(',')
if (mov[0] != 'null') {
	for (const vd of mov){
  tR +=`<video width="600" height="400" src="${vd}" controls></video>`;
  };
} else
{ tR += ``};
%>
  </details>
  </center>

\
	


<%*
const sc = String(tp.frontmatter.screenshots).split(',')
for (const elem of sc) {
  tR +=`![|192x108](${elem})`;
};
%>



Review: 

>




<details>
<summary></summary>
<%tp.user.add(tp.frontmatter.id, tp.frontmatter.screenshots)%>
</details>
  
```

I have a ton of ideas, and I'll gradually implement them as I learn more about js and the capabilities of Templater and dataview. Little video how it works now. I'll this post with future updates. There's a lot of work ahead with Templater to create a beautiful and user-friendly note page.

https://user-images.githubusercontent.com/22260439/227794424-2808d179-b625-48a3-8350-c2fbedc33d9b.mp4

upd obsidian-media-db-plugin with few change. Just install plugin and change main.js in .obsidian\plugins\obsidian-media-db-plugin It will work until you update plugin. Source code with nessesary changes https://github.com/talraash/obsidian-media-db-plugin

[main.zip](https://github.com/SilentVoid13/Templater/files/11082724/main.zip)

**Comment by welpdx** [comment]
Interesting!
Love the diagram and I cant wait to see how you get it to have those two columns. 
nice work on the background img

**Reply by talraash** [reply]
> I cant wait to see how you get it to have those two columns.

Looks like it's impossible task for me. I tryed  iframes and html/css. It's look nice only in preview mode. In editing mode it's awful from esthetic view and usability. So... looks like i'll stop on this.

[2023-03-27 22-59-38.webm](https://user-images.githubusercontent.com/22260439/228058554-f85519c5-213e-4967-90cd-a76d14a1c57a.webm)

**Reply by welpdx** [reply]
that looks really cool!

(btw, you can use sharex to easily record video, convert it to gif, upload it to imgur, and share a link)

**Comment by Mastecker** [comment]
I love your idea. Unfortunate that its not possible the way you imagined it...
Maybe you could work with tables?

Would be very interested on how to integrate things like the Steam-video and the developer name without tingling with the plugin... Do you have an idea? I mean the Steam API gets used and the information should therefore be requestable....

**Reply by talraash** [reply]
> Maybe you could work with tables?

I used css+html. Like the tables in markdown it looks good in preview mode but bad in edit mode which is where I spend 90% of the time.

>Would be very interested on how to integrate things like the Steam-video and the developer name without tingling with the plugin... Do you have an idea?

For example we can take developers from id.data.developers from request to steam [Steam_web_api](https://developer.valvesoftware.com/wiki/Steam_Web_API) The plan is to get a lot more information from steam.
So it's easier to write everything from scratch in a simple steam-only script than to modify an existing plugin. Add getting a list of user's games etc. and automatic creation of notes on this list with some metadata. Of course, I will post everything here script and my vision of templates. When it's ready...

**Comment by OPEN1004MIND** [comment]
I can create a game title md file for dark forces like the demo sample, but I don't know how to create an md file like the sample.

I'm not sure if you name them app_add.js and cs_.md, can you explain how the files work together? 
And what is the name of the template file?

The main.js I attached, do I overwrite it after installing media-db-plugin? 
Or just copy them under snipper?

**Reply by talraash** [reply]
Hi you need just two files beside modifyed  media db plugin. Some js script( i call it add.js) that create or edite .css you need set proper path to you vault in this script in const filePath variable. Also make sure you turn on this main.css in obsidian Options>Appereance after first game add(one time required), and this js file must be in you templater "Script file folders location". And you need template with any name just set proper template for games in media db plugin settings. I checked it work now as intend but not for all games, it requires additional fix in media db plugin now.

P.S. Yes you need modifyed media db plugin pre compiled from me(you can replace main.js in existing plugin) or you can compile it from source. I haven't actually used a third party plugin for a long time. I wrote my own implementation of data scraping from valve servers. But there is no time to arrange everything in a form that could be shared with the community.

**Comment by Oddward** [comment]
There's a cool plugin called [Game search](https://github.com/CMorooney/obsidian-game-search-plugin) that pulls various game metadata from [RAWG](https://rawg.io) (data-focused game platform with profiles that can add your games from Steam and other sources, API requires a key free with a signup). It has its own templating tags too. Maybe you could utilize that for a games catalogue

I was doing that with the basic Obsidian Templates, but the rest of the content I gotta get manually ^^; Maybe Templater can complete it

*It was just updated to support Steam sync directly ~

**Reply by talraash** [reply]
Thanks, RAWG good for non pc games. I'll make it the second source option in my script.
